src/enochecker/utils.py

```diff
@@ -204,8 +204,15 @@ def __init__(
             self.logger = logger
         else:
             self.logger = utilslogger
+        self.recv: None = self.recv_patched  # type: ignore
         self.timeout_fun: Optional[Callable[[], float]] = timeout_fun
 
+    def recv_patched(self, *args: Tuple[Any, ...], **kwargs: Dict[str, Any]) -> bytes:
```

Comment:
Why not just `def self.recv`? If for doc strings, there's a decorator for that

Reply:
I wanted pwnlib functions to call the new recv too and thought that overwriting it in a subclass with `def recv(self, ..)` would only cause other functions from SimpleSocket to use the new version and functions from pwnlib's tubes to use the old one. A simple test shows that the behavior is equivalent though:
```
class Base:
    def foo(self):
        print("Base foo")

    def bar(self):
        print("Base bar")
        self.foo()

class Sub(Base):
    def foo(self):
        print("Sub foo")

base = Base()
base.bar() # -> base bar -> base foo

sub = Sub()
sub.bar() # -> base bar -> sub foo
```

I was also wondering if we might need to overwrite `read` as well, but it is defined as a [wrapper](https://github.com/Gallopsled/pwntools/blob/78c6b8c2a9746e96f7393e45a51fc30d5b523445/pwnlib/tubes/tube.py#L1445) and not a direct copy so it will inherit the new behavior.

Reply:
What is that decorator called? Otherwise I would do `__doc__ = super().__doc__`